src/cli.rs

```diff
@@ -0,0 +1,191 @@
+use std::convert::From;
+use std::env;
+use std::fs::{File, OpenOptions};
+use std::io::{self, Read, Write};
+use std::path::{Path, PathBuf};
+
+use clap::builder::PossibleValue;
+use clap::{arg, command, value_parser, Arg, ArgAction, ValueEnum};
+use glob::glob;
+
+extern crate minifier;
+use minifier::{css, js, json};
+
+pub struct Cli;
+
+impl Cli {
+    pub fn init() {
+        let matches = command!()
+            .arg(
+                Arg::new("FileType")
+                    .short('t')
+                    .long("type")
+                    .help(
+                        "File Extention without dot. This option is optional.
+If you don't provide this option, all input files
+type will detect via extension of input file.
+",
+                    )
+                    .required(false)
+                    .value_parser(value_parser!(FileType)),
+            )
+            .arg(
+                Arg::new("output")
+                    .short('o')
+                    .long("out")
+                    .help("Output file or directory (Default is parent dir of input files)")
+                    .required(false)
+                    .value_parser(value_parser!(PathBuf)),
+            )
+            .arg(
+                arg!(<FILE>)
+                    .help("Input Files...")
+                    .num_args(1..)
+                    .value_parser(value_parser!(String))
+                    .action(ArgAction::Append),
+            )
+            .get_matches();
+        let args: Vec<&str> = matches
+            .get_many::<String>("FILE")
+            .unwrap_or_default()
+            .map(|v| v.as_str())
+            .collect::<Vec<_>>();
+        let ext: Option<&FileType> = matches.get_one::<FileType>("FileType");
+        let out: Option<&PathBuf> = matches.get_one::<PathBuf>("output");
+        for i in args {
+            for entry in glob(i).expect("Failed to read glob pattern") {
```

Comment:
When would globs not be expanded by the shell?

Reply:
Thank you for your feedback. I included the glob pattern matching in the code to handle cases where the shell might not expand globs by default, such as when running on Windows or certain environments. However, if this isn't needed, I'm happy to update the code.

Reply:
The problem with that approach is that if there is a star on linux, then it's not supposed to be expanded. I'd prefer if you removed it.

Reply:
I have removed it.